File: CodeGuessing/CG76/golomb.go
var table = make(map[int]int)

func entry(x int) int {
	table[1] = 1
	val, ok := table[x]
	if ok {
		return val
	}
	table[x] = 1 + entry(x-entry(entry(x-1)))
	return table[x]
}

File: CodeGuessing/CG76/golomb.java
import java.util.HashMap;

class Main {
    private static int entry(int x) {
        final HashMap<Integer, Integer> table = new HashMap<Integer, Integer>();
        table.put(1, 1);

        if(table.get(x) != null) {
            return table.get(x);
        }

        int y = 1 + entry(x - entry(entry(x - 1)));
        table.put(x, y);
        return y;
    }
}
